docs/terminal/shell-integration.md

```diff
@@ -91,6 +91,35 @@ When using shell integration, it has a "quality" associated with it that declare
 
 To view the shell integration quality, hover the terminal tab. Optionally, select **Show Details** on the hover to view more detailed information.
 
+## IntelliSense (Preview)
+
+### Background
+
+IntelliSense, powered by shell integration `setting(terminal.integrated.shellIntegration.enable)`, can be enabled with `setting(terminal.integrated.suggest.enabled)`.
+
+By default, on request of completions `kb(workbench.action.terminal.requestCompletions)` files, folders, commands, arguments, and options are provided.
+
+![A user has typed git checkout in a terminal and requests completions. The completion list shows branch names like main and merogge/terminal-section](images/shell-integration/terminal-suggest.png)
+
+Commands are sourced from [Fig specs](https://github.com/withfig) and per-shell builtin functions (for `pwsh`, `bash`, `zsh`, and `fish`) and are validated against the `$PATH` to ensure they exist. On Windows, the particular set of executables can be configured with `setting(terminal.integrated.suggest.windowsExecutableExtensions)`.
+
+There can be multiple IntelliSense providers, which can be changed with `setting(terminal.integrated.suggest.providers)`.
+
+### Keyboard navigation
+
+By default, `Tab` inserts the suggestion. Once navigation of the list has occurred, `Enter` will similarly insert the suggestion. This can be configured with `setting(terminal.integrated.suggest.selectionMode)`.
+
+To both insert and run the completion in the terminal on acceptance, configure `setting(terminal.integrated.suggest.runOnEnter)`.
+
+IntelliSense is triggered when typing, which can be disabled with `setting(terminal.integrated.suggest.quickSuggestions)`. Intellisense can also be triggered when certain characters are typed, such as `/`, which can be configured with `terminal.integrated.suggest.suggestOnTriggerCharacters`.
```

Comment:
If I disable triggering by typing, am I them completely disabling it or do I need to manually trigger them? Not clear when reading this sentence. 
It can also be triggered when typing a character. Is this then the alternative to typing or an extra thing on top of manually triggering a completion?